src/api/api.service.ts

```diff
@@ -231,23 +231,30 @@ export class ApiService {
     createEndpointsDto: CreateEndpointDto[],
     queryRunner: QueryRunner,
   ) {
-    const newEndpoints = createEndpointsDto.map((createEndpointDto) =>
-      this.endpointsRepo.create({
-        ...createEndpointDto,
-        apiId,
-      }),
-    );
-    // Check for duplicate endpoints...
     const endpointSet = new Set();
     const duplicateEndpoints = [];
-    for (const endpoint of newEndpoints) {
+    const endpointsToSave = [];
+    for (const createEndpointDto of createEndpointsDto) {
+      const endpoint = this.endpointsRepo.create({
+        ...createEndpointDto,
+        apiId,
+      });
       const endpointKey = `${endpoint.method} ${endpoint.route}`;
       if (endpointSet.has(endpointKey)) {
```

Comment:
@devynlab i think this is better    
      if (!endpointSet.has(endpointKey)) {
          endpointSet.add(endpointKey);
          endpointsToSave.push(endpoint);
        }

Reply:
Done